include/deal.II/sundials/ida.h

```diff
@@ -0,0 +1,653 @@
+     * call to solver_should_restart().
+     *
+     * By default solver_should_restart returns false. If the user needs to
+     * implement, for example, local adaptivity in space, he or she may assign
+     * a different function to solver_should_restart() that performs all mesh
+     * changes, transfers the solution and the solution dot to the new mesh, and
+     * returns true.
+     *
+     * During reset(), both y and yp are checked for consistency, and according to
+     * what was specified as ic_type (if t==initial_time) or reset_type (if
+     * t>initial_time), yp, y, or both are modified to obtain a consistent set
+     * of initial data.
+     *
+     * @param[in] t  The new starting time
+     * @param[in] h  The new (tentative) starting time step
+     * @param[in,out] y   The new (tentative) initial solution
+     * @param[in,out] yp  The new (tentative) initial solution_dot
+     */
+    void reset(const double &t,
+               const double &h,
+               VectorType &y,
+               VectorType &yp);
+
+    /**
+     * Reinit vector to have the right size, MPI communicator, etc.
+     */
+    std::function<void(VectorType &)> reinit_vector;
+
+    /**
+     * Compute residual.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (IDAReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const double t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      VectorType &res)> residual;
+
+    /**
+     * Compute Jacobian.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (IDAReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const double t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      const double alpha)> setup_jacobian;
+
+    /**
+     * Solve linear system.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (IDAReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const VectorType &rhs, VectorType &dst)> solve_jacobian_system;
+
+    /**
+     * Store solutions to file.
+     */
+    std::function<void (const double t,
+                        const VectorType &sol,
+                        const VectorType &sol_dot,
+                        const unsigned int step_number)> output_step;
```

Comment:
I think you wanted to update the description (and maybe name) to state when the function will be called and what it is supposed to do.

Reply:
done.